app/src/main/AndroidManifest.xml

```diff
@@ -33,13 +34,23 @@
             android:name=".HelpActivity"
             android:label="@string/activity_label_help" />
 
+        <service
+            android:name=".ShutdownReceiverService"
+            android:label="@string/service_label_shutdown" />
+
         <receiver
             android:name=".BootReceiver"
             android:enabled="true"
             android:exported="true">
             <intent-filter android:priority="9999">
                 <action android:name="android.intent.action.BOOT_COMPLETED" />
             </intent-filter>
+            <intent-filter android:priority="9999">
+                <action android:name="android.intent.action.MY_PACKAGE_REPLACED" />
```

Comment:
I'm not sure why you added the two additional intents. If I understand correctly, this would trigger the boot receiver when the app gets updated or device policies change which would then restore old ringer / DND values.